src/Tasks/CodeTaskFactory.cs

```diff
@@ -791,18 +791,24 @@ private Assembly CompileInMemoryAssembly()
                 // Our code generation is complete, grab the source from the builder ready for compilation
                 string fullCode = codeBuilder.ToString();
 
+                string tempDirectory = FileUtilities.TempFileDirectory;
+                string fileName = Guid.NewGuid().ToString() + ".txt";
+                string outputPath = Path.Combine(tempDirectory, fileName);
+
+                // Embed generated file in the binlog
+                _log.LogIncludeGeneratedFile(outputPath, fullCode);
+
+                // Log the location of the code file in binlog
+                _log.LogMessageFromResources(MessageImportance.Low, "CodeTaskFactory.FindSourceFileInBinlogAt", outputPath);
```

Comment:
What's the motivation for this separate message? I'm concerned about having it in a diag text log (when no binlog was attached).

Reply:
+1, also the message can be seen as deceptive because, unless `MSBUILDLOGCODETASKFACTORYOUTPUT` is set or there are compilation errors, it will log a non-existent path.

Reply:
Thanks @rainersigwald and @ladipro! These are good points. 

The motivation for this separate message was to tell the user where to search for the file in the binlog. I wanted the file path to be same as for the compilation file. But as @ladipro mentioned it can be deceptive. And if the binlog is not attached that is also deceptive as @rainersigwald said. 

Given the feedback, I think there should be a generic path for this file. What do you think about {project name}/{some guid}-{task name}-compilation-file.txt?

Reply:
> What do you think about {project path}/{some guid}-{task name}-compilation-file.txt?

Sounds great. I think one scenario where the path under which the file is embedded in binlog is actually relevant is when reconstructing the project skeleton from binlog (didn't Kirill have an experimental tool for that?) I would expect that using paths under the project path would lead to good experience there.

Reply:
Some repos maintain very strong requirements that no build outputs be placed in the source folder, so I don't think putting a "log" file next to the project is a good idea.

If it's a separate message type though, we should be able to make "double click on it in the viewer" just open the right file.

Reply:
Let's see, I may not have understood the proposal. We're not talking about changing the path where actual files are dropped, only the fake paths of code fragments that otherwise exist only in memory and we want to persist them in binlog, is that correct?

Reply:
Ah, got it. Sounds fine then!

Reply:
Great! I'm going to implement it.

Reply:
It was challenging to get the project directory. I added `IHasProjectFullPath` interface to get it from the logger. I'll update the PR description.